core/src/com/unciv/ui/mapeditor/MapEditorScreen.kt

```diff
@@ -11,14 +11,14 @@ import com.badlogic.gdx.utils.Array
 import com.unciv.logic.MapSaver
 import com.unciv.logic.map.TileInfo
 import com.unciv.logic.map.TileMap
-import com.unciv.models.ruleset.RulesetCache
 import com.unciv.models.translations.tr
-import com.unciv.ui.newgamescreen.GameParametersPreviousScreen
 import com.unciv.ui.newgamescreen.GameSetupInfo
+import com.unciv.ui.newgamescreen.PreviousScreenInterface
 import com.unciv.ui.utils.*
 
-class MapEditorScreen(): GameParametersPreviousScreen() {
-    override val ruleset = RulesetCache.getBaseRuleset()
+class MapEditorScreen(): PreviousScreenInterface, CameraStageBaseScreen() {
```

Comment:
Why do we need a PreviousScreenInterface? We can just create the MapEditorScreen with GameSetupInfo and Ruleset constructor parameters, that feels much cleaner to me.

Reply:
I agree it looks not so clean for me. I introduced `PreviousScreenInterface` mainly because of reusability of `PlayerPickerTable`. We used it in two cases - during new game creation and in the map editor. `PlayerPickerTable class `is assumed to get `rightSightButton `from previous screen,  however this is the case only for new game screen, a PickerScreen ancestor:

`core/src/com/unciv/ui/newgamescreen/PlayerPickerTable.kt :`
` 53:       previousScreen.setRightSideButtonEnabled(gameParameters.players.size > 1)`

When I tried to use `PlayerPickerTable` in the Map Editor screen, which is `CameraBackStageScreen `ancestor and don't have `rightSightButton`, `topTable `and others, so I get errors.
The idea of `PreviousScreenInterface `is to provide ability to use `PickerScreen `childs and `CameraBackStageScreen `childs both as an argument for PlayerPickerTable.

The similar logic behind GameOptionsTable, which is used by both NewGameScreen: PickerScreen() and MapEditorScreen: CameraBackStageScreen ()

I am sure there should be more elegant solution, but for the moment I couldn't think of it. Any help would be appreciated :)

Reply:
We can add setRightSideButtonEnabled as a function for the base PickerScreen instead, and thus avoid an extra class